### PR TITLE
Unify cesium.com deploy

### PR DESCRIPTION
**Pull request by ggetz**
We currently have a fair amount of deploy code that lives only in the `cesium.com` branch. The `cesium.com` branch is the "production" branch; When we push to it as part of the release process, travis CI will deploy files to the website for use in Sandcastle, documentation, cesium viewer, and a list of releases.

However, due to this separate deployment code across branches, it can become a chore to merge in `main` given any updates to `.travis.yml` or the gulpfile such as when updating dev packages. This PR seeks to unify the logic as much as possible, pulling in the custom deploy logic from that branch and conditionally running it based on Travis environment variables.

You can test the "production" paths locally by setting a local `TRAVIS_BRANCH`  environment variable to `"cesium.com"`

- Unified `.travis.yml`, moving any complex logic to a script, as recommended per the travis CI documentation
- Fixes https://github.com/CesiumGS/cesium/issues/10267 by disabling tests for the cesium.com branch
- Remove redundant build steps in the cesium.com branch
- Renamed `scripts`  to `travis` to be consistant with our other repos
- Updated `gulpfile.js`  such that it should be the same in both branches. There are now conditional paths where the differences were, which are run only in travis when the branch name matches `cesium.com`.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ggetz!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/unify-cesium.com-deploy/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: gulpfile.js [hunk 1]
```diff
@@ -416,11 +421,31 @@ export function buildDocs() {
...
   console.log("Listening for changes in documentation...");
   return gulp.watch(sourceFiles, buildDocs);
 }
 
+function combineForSandcastle() {
+  const outputDirectory = join("Build", "Sandcastle", "CesiumUnminified");
+  return build({
+    minify: false,
+    removePragmas: false,
+    outputDirectory: outputDirectory,
```
Comment (sanjeetsuhag) [inline gulpfile.js:433]: Does the outputDirectory argument matter here? It's not passed to buildCesium from build.

**Review by sanjeetsuhag — COMMENTED** [review]
Thanks @ggetz, this will be a great QoL change around release time.

**Comment by ggetz** [comment]
Thanks @sanjeetsuhag! This should be ready fro another look.

**Review by sanjeetsuhag — APPROVED** [review]
Looks good! Thanks @ggetz!

**Pull request merged**